motion_tracker/utils/image_generator.py

```diff
@@ -0,0 +1,193 @@
+    img_metadata = pd.read_csv('work/alov/parsed_bb.csv')
+
+    empty_img_accumulator = np.zeros([0, output_width, output_height, 3]).astype('uint8')
+    empty_box_accumulator = np.zeros([0, 4])
+
+    while True:
+        my_output = {'start_img': empty_img_accumulator,
+                     'start_box': empty_box_accumulator,
+                     'end_img': empty_img_accumulator,
+                     'end_box': empty_box_accumulator}
+        while my_output['start_img'].shape[0] < batch_size:
+            img_row = img_metadata.sample(1)
+
+            start_img0 = cv2.imread(raw_image_dir + img_row.filename_start.values[0])
+            end_img0 = cv2.imread(raw_image_dir + img_row.filename_end.values[0])
+
+            
+            start_img_coords = Coords(0, 0, start_img0.shape[1], start_img0.shape[0])
+            end_img_coords = Coords(0, 0, end_img0.shape[1], end_img0.shape[0])
+            start_box_coords = Coords(img_row.x0_start, img_row.y0_start, 
+                                      img_row.x1_start, img_row.y1_start)
+
+            end_box_coords = Coords(img_row.x0_end, img_row.y0_end, 
+                                    img_row.x1_end, img_row.y1_end)
+            start_img, start_box = crop_and_resize(start_img0, start_img_coords,
+                                                   start_box_coords, output_width,
+                                                   output_height, random_crop=False)
+            start_img, start_box = set_array_dims(start_img, start_box)
+
+            for image_crop in range(crops_per_image):
+                end_img, end_box = crop_and_resize(end_img0, end_img_coords,
+                                                   end_box_coords, output_width,
+                                                   output_height, random_crop=True)
+                end_img, end_box = set_array_dims(end_img, end_box)
+                my_output['start_img'] = np.concatenate([my_output['start_img'],
+                                                         start_img])
+                my_output['start_box'] = np.concatenate([my_output['start_box'],
+                                                         start_box])
+                my_output['end_img'] = np.concatenate([my_output['end_img'],
+                                                       end_img])
+                my_output['end_box'] = np.concatenate([my_output['end_box'],
+                                                       end_box])
+
+        # Limit output arrays from being larger than batch_size
+        for field in my_output:
+            my_output[field] = my_output[field][:batch_size]
+        yield my_output
+
+if __name__ == '__main__':
+    my_gen = master_generator(crops_per_image=10, batch_size=50)
+    batch = next(my_gen)
+    for i in range(batch['start_img'].shape[0]):
+        show_single_stage(batch['start_img'][i], batch['start_box'][i])
+        show_single_stage(batch['end_img'][i], batch['end_box'][i])
```

Comment:
So.... this should work. I was able to see 10 sets of an image pair (e.g. 20 images) before seeing another 10 sets of an image pair, and they corresponded to both `alov` and `imagenet`. 😆 

I didn't make it to any actual refactoring of the image_cropping file, so that's still on my list. After working through it some in trying to debug today, I don't know that there's a lot to do. Also, given the errors we've run into, I think it could be worthwhile in the future/long term to throw some asserts/checks in there that give warnings as to what's wrong. For example, at one point I was getting cropping coords that were greater than the image, and so the cropping coords were restricted to be the max of the image -1. This is definitely the right thing to do coding wise, but it might be nice to throw some kind of warning in there to note that that is going on. It also might be something that is much easier to find if you've worked with images before, leaving any kind of warnings less useful.